### PR TITLE
add log softmax problem

### PR DESCRIPTION
**Pull request by MKFMIKU**
add a real internview problem that discuss the numerical instability of softmax function

### CODE REVIEW

File: Problems/29_log_softmax/learn.html
Comment (Open-Deep-ML) [inline Problems/29_log_softmax/learn.html:1]: what are your thoughts on this learn section?

<h2>Understanding Log Softmax Function</h2>

<p>The log softmax function is a numerically stable way of calculating the logarithm of the softmax function. The softmax function converts a vector of arbitrary values (logits) into a vector of probabilities, where each value lies between 0 and 1, and the values sum to 1. The softmax function is given by:</p>

\[
\text{softmax}(x_i) = \frac{e^{x_i}}{\sum_{j=1}^n e^{x_j}}
\]

<p>However, directly applying the logarithm to the softmax function can lead to numerical instability, especially when dealing with large numbers. To prevent this, we use the log-softmax function, which incorporates a shift by subtracting the maximum value from the input vector:</p>

\[
\text{log\_softmax}(x_i) = x_i - \max(x) - \log\left(\sum_{j=1}^n e^{x_j - \max(x)}\right)
\]

<p>This formulation helps to avoid overflow issues that can occur when exponentiating large numbers. The log-softmax function is particularly useful in machine learning for calculating probabilities in a stable manner, especially when used with cross-entropy loss functions.</p>

**Review by Open-Deep-ML — COMMENTED** [review]
see my comments on the learn section, also could you change the problem number for the folder name to 39 (still adding legacy problems from https://www.deep-ml.com/ and I want the IDs to be consistent with the site.

File: Problems/29_log_softmax/learn.html
Comment (MKFMIKU) [reply Problems/29_log_softmax/learn.html:1]: sounds good to me. updated the description.

**Review by Open-Deep-ML — APPROVED** [review]
